solutions/day-28/src/components/Post.js

```diff
@@ -0,0 +1,69 @@
+import { useRef, useState } from "react";
+import "../styles/Post.css";
+
+const Post = ({ data, onDelete, onEdit }) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const textareaRef = useRef();
+
+  const onEditPost = () => {
+    setIsEditing(true);
+  };
+
+  const onSaveEditedPost = () => {
+    data.tweet = textareaRef.current.value;
+    setIsEditing(false);
+  };
+
+  const onCancelEditPost = () => {
+    setIsEditing(false);
+  };
+
+  if (isEditing) {
+    return (
+      <div className="post">
+        <textarea defaultValue={data.tweet} ref={textareaRef} />
+        <br />
+        <button className="save-button" onClick={onSaveEditedPost}>
+          Save
+        </button>
+        <button className="cancel-button" onClick={onCancelEditPost}>
+          Cancel
+        </button>
+      </div>
+    );
+  }
+
+  return (
+    <div className="post">
+      <div className="user-info">
+        <i className="fas fa-user" />
+        <h4>{data.fullName}</h4>&nbsp;
+        <span>{data.username}</span>
+      </div>
+      <div>
+        <p>{data.tweet}</p>
+      </div>
+      <div className="post-detail">
+        <div>
+          <i className="far fa-edit" onClick={onEditPost} />
+          <i
+            className="far fa-trash-alt"
+            onClick={() => {
+              onDelete(data);
+            }}
+          />
```

Comment:
same feedback as https://github.com/jhrasco-ct/30-Days-Of-React/pull/22#discussion_r1045701227

Reply:
Updated. Thanks, @ianabalus.